src/DefaultRequestAdapterBuilder.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Microsoft\Kiota\Bundle;
+
+use GuzzleHttp\ClientInterface;
+use Microsoft\Kiota\Abstractions\Authentication\AnonymousAuthenticationProvider;
+use Microsoft\Kiota\Abstractions\Authentication\AuthenticationProvider;
+use Microsoft\Kiota\Abstractions\Serialization\ParseNodeFactory;
+use Microsoft\Kiota\Abstractions\Serialization\SerializationWriterFactory;
+use Microsoft\Kiota\Http\Middleware\Options\ObservabilityOption;
+
+class DefaultRequestAdapterBuilder
+{
+    private AuthenticationProvider $authenticationProvider;
+    private ?ParseNodeFactory $parseNodeFactory = null;
+    private ?SerializationWriterFactory $serializationWriterFactory = null;
+    private ?ClientInterface $guzzleClient = null;
+    private ?ObservabilityOption $observabilityOption = null;
+
+    public function __construct()
+    {
+        $this->authenticationProvider = new AnonymousAuthenticationProvider();
+    }
+
+    public function withAuthenticationProvider(AuthenticationProvider $authenticationProvider): DefaultRequestAdapterBuilder
```

Comment:
aren't we missing a bunch of doc comments here?

Reply:
added in https://github.com/microsoft/kiota-bundle-php/pull/1/commits/611fa4b1fa2101a696e0273904a8a30e12c7f1df